src/App.tsx

```diff
@@ -1,9 +1,23 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+
+  const filterMovies = (movie: Movie, field: string) => {
+    const fieldForFilter = field.toLowerCase();
+    const movieTitle = movie.title.toLowerCase();
+    const movieDescription = movie.description.toLowerCase();
+
+    return movieTitle.includes(fieldForFilter)
+      || movieDescription.includes(fieldForFilter);
+  };
+
+  const visibleMovies = moviesFromServer
+    .filter(movie => filterMovies(movie, query));
```

Comment:
it's better to save movies in the state and remove visibleMovies variable. After filtering just update the state with movies